### PR TITLE
VERSION => version.py

### LINKED ISSUE #345: compile issues with pyinstaller
**Objective:** compile an app (using pyinstaller) that has the smart_open module.

**Issue:** I think the "VERSION" file isn't importing correctly.  Once the app compiles, if you start it in the command line, the error will trigger.

**Error:** Here's the error message I'm seeing when running the app

```
Traceback (most recent call last):
  File "splitter_129.py", line 26, in <module>
    import smart_open
  File "<frozen importlib._bootstrap>", line 971, in _find_and_load
  File "<frozen importlib._bootstrap>", line 955, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 665, in _load_unlocked
  File "c:\python360564\lib\site-packages\PyInstaller\loader\pyimod03_importers.py", line 627, in exec_module
    exec(bytecode, module.__dict__)
  File "smart_open\__init__.py", line 39, in <module>
  File "smart_open\__init__.py", line 35, in _get_version
  File "smart_open\smart_open_lib.py", line 307, in open
  File "smart_open\smart_open_lib.py", line 512, in _shortcut_open
FileNotFoundError: [Errno 2] No such file or directory: 'C:\\Users\\domccorm\\AppData\\Local\\Temp\\_MEI118122\\smart_open\\VERSION'
```

**Environment:** Windows 10, Python360564

It looks like someone else ran into this issue and posted within SO: https://stackoverflow.com/questions/57482672/filenotfounderror-errno-2-when-running-pyinstaller-executable

I'm not sure if the suggestions/answers on SO work -- I've been struggling and testing (trial and error) to see if I can fix it

Thanks so much,
-Doug Jr.

### PR DESCRIPTION
**Pull request by rs-trevor**
Python has a rad feature where you can import zip files directly in your python path. PEX (https://pypi.org/project/pex/) leverages it.  Twitter has made a whole CI/CD around it.

`PYTHONPATH=your_deps.zip python -c print("yay")`

Unfortunately reading the flat VERSION file doesn't work with it.  Can't relative read from a zip file.  This PR is to get around this issue.  It involves making VERSION into a python file, which means your lib can report it's version back without hitting the disk which is nice.  I believe I updated your deployment scripts too.  

Let me know if there is a more preferable way to deal with this.

Fixes #345.

### CODE REVIEW

**Comment by douglasmccormickjr** [comment]
This solves a pyinstaller issue when trying to run the compiled program.  The issue I encountered is documented here: https://github.com/RaRe-Technologies/smart_open/issues/345

Hope this helps a bit.

Thanks,
-Doug Jr.

**Comment by mpenkov** [comment]
@rs-trevor The tests seem to be failing. Could you please look into it?

**Comment by mpenkov** [comment]
It's unexpected because it appears to be a direct consequence of your changes.

Recent builds without your changes (e.g. [this one](https://travis-ci.org/RaRe-Technologies/smart_open/builds/577228030?utm_source=github_status&utm_medium=notification)) do no exhibit the same problem.

Can you please investigate?

**Comment by rs-trevor** [comment]
you're absolutely right, i should have commented.  it's an issue with including python deps that's a bit hard to follow in the build scripts -- would you mind if I added a Makefile to the project? It would make some of the build dependencies a bit clearer.

**Comment by mpenkov** [comment]
I'd prefer to avoid Makefiles.

You can use them to diagnose the problem, but please don't include it in the final PR.

**Comment by rs-trevor** [comment]
The build scripts are pretty difficult to work with, Makefiles generally help solving that problem.

**Comment by piskvorky** [comment]
What's going on, why was this closed?

**Comment by mpenkov** [comment]
@piskvorky I couldn't write to the branch behind this PR, so I opened a new one.

**Comment by piskvorky** [comment]
Great, thanks @mpenkov !
